### DISCUSSION TITLE
Migrate on-premises installation to docker

### DISCUSSION THREAD

**Discussion by ghost** [amidaware/tacticalrmm#1558] [Q&A]
At the moment we are using Tactical RMM in a Debian 12 VM on Proxmox. 
Unfortunately, there seems to be an unsolved bug with qemu-guest-agent on Debian (and maybe Ubuntu) that randomly freezes the VM with no solution except an hard reset of the VM.

We'd like to try moving our on-premises installation on a docker container on Alpine Linux. 
Can the backup/restore script be used on this scenario?

**Comment by silversword411** [comment]
I'm going to test deb12vm on proxmox with qemu-guest-agent, see if I see the same.

> Can the backup/restore script be used on this scenario?

No, but you can look at the script and see how the database backups happen, run that manually in the right container and then you'll have to manually move them over, and manually import them again.

We would only be able to help further under paid commercial support.
https://docs.tacticalrmm.com/support/#commercial-support

**Reply by ghost** [reply]
Thank you. Adding more information:
- Other VMs on pve (alpine) + qemu agent: no freeze
- Other VMs on pve (debian 11) + qemu agent: no freeze
- Tactical RMM on pve (debian 11) + qemu agent: random freeze
- Tactical RMM on pve (debian 12) + qemu agent: random freeze
- Tactical RMM on pve (debian 12) without qemu agent: random freeze

**Reply by silversword411** [reply]
> * Tactical RMM on pve (debian 11) + qemu agent: random freeze

You mean this? This is my prod deb11 server, and I've never had a freeze. This was built 5/30, so it's got over a month of constant run/use


<img width="525" alt="2023-07-10_111854 - brave_proxmox" src="https://github.com/amidaware/tacticalrmm/assets/13395348/d5479c93-a02d-44ff-81cf-2e4349ca2193">

**Reply by ghost** [reply]
Yep, it was probably due to some kind of kernel incompatibility (5.15) on pve with Intel N6005. 
Upgraded yesterday to kernel 6.2 and we'll see. So far, it seems stable. I have no idea why it happens only on Debian VMs and not with other VMs or with the hypervisor.

**Reply by silversword411** [reply]
Oh, yeah that's a pretty slow proc, don't use any of those.
